src/indexes/text/text_index.cc

```diff
@@ -35,6 +35,16 @@ std::optional<std::shared_ptr<Postings>> AddKeyToPostings(
     current_schema_->GetMetadata().num_unique_terms++;
     postings = std::make_shared<Postings>();
   }
+
+  // Track metadata before inserting
```

Comment:
I'm not sure this is the fix we want to make. The issue may be testing related - we have tests that directly test the posting object API. InsertKey and RemoveKey fetch and access the thread-local schema object that wouldn't have been set. This is another indicator that we need to be careful with (and maybe rethink) this global thread-local pattern.